### PR TITLE
CVE-2010-1157

### PR DESCRIPTION
**Pull request by rxv3774**
Researched CVE-2010-1157 vulnerability.

### CODE REVIEW

File: cves/CVE-2010-1157.yml [hunk 1]
```diff
@@ -91,7 +84,7 @@ upvotes_instructions: |
   upvotes to each vulnerability you see. Your peers will tell you how
   interesting they think this vulnerability is, and you'll add that to the
   upvotes score on your branch.
-upvotes: 4
+upvotes:
```
Comment (zxw4320) [inline cves/CVE-2010-1157.yml]: upvotes: 3

File: cves/CVE-2010-1157.yml [hunk 2]
```diff
@@ -149,15 +140,11 @@ interesting_commits:
     emerging themes?
     If there are no interesting commits, demonstrate that you completed this
     section by explaining what happened between the VCCs and the fix.
-  answer: |
-    The git history for this directory only goes back to a branching of Tomcat 5.5.x. Between
-    that commit and the fix for the issue, there is a commit that mentions a fix for session fixation.
-    Apparently session fixation is an attack that alllows a hacker to hijack a user session. The commit
-    aimed to prevent this by providing the option to change session ID upon authentication.
+  answer: Between the VCC's and the fix, there did not appear to be any interesting commits.
```
Comment (zxw4320) [inline cves/CVE-2010-1157.yml]: I understand there was nothing of interest, but this does not show you completed this step. Provide an explanation of what happened between the VCC and the fix as stated.

File: cves/CVE-2010-1157.yml [hunk 3]
```diff
@@ -220,7 +208,7 @@ mistakes:
...
     engineering industry would find interesting.
   answer: |
-    It seems that this vulnerability was an oversight with regards to how the configuration and authentication code
-    interacted. The fix was to provide a default realm name that did not contain sensitive data, instead of 
-    generating a string that used the actual hostname and port. What lead to this vulnerability was most likely
-    a simple ignorance to the value of a server's hostname information.
+    This vulnerability seemed to be a result of a design mistake. The developers
+    should have seen that giving out the IP address of the Tomcat machine so easily
+    was a risk. Overall the team decided to scrap the output and switched
+    it over to something that was much more safe (a default hard-coded string).
```
Comment (zxw4320) [inline cves/CVE-2010-1157.yml]: Switched it over to something safer vs something that was much more safe. Clearer and easier to read.

File: cves/CVE-2010-1157.yml [hunk 4]
```diff
@@ -44,45 +44,38 @@ description_instructions: |
...
   keep too.
 description: |
-  A portion of Tomcat that handles http authentication via DIGEST and BASIC strategies had the potential to allow
-  for the viewing of the local name or IP address of the server running Tomcat. The header for the authentication 
-  via these strategies includes a <realm-name> element.
-  If a <realm-name> element is specified for the application in one of the configuration files, it will be used. 
-  However, if a <realm-name> is not specified then Tomcat will generate a realm name and
-  in some circumstances this can expose the local host name or IP address of the machine running Tomcat. 
-
+  In an optional header, a user is able to obtain and expose the IP address
```
Comment (kwc2894) [inline cves/CVE-2010-1157.yml]: I know you said it was in the optional header but what did the user do with the header? (click, enter a command, ect)

File: cves/CVE-2010-1157.yml (hunk 1 above)
Comment (kwc2894) [inline cves/CVE-2010-1157.yml]: Upvotes 2

File: cves/CVE-2010-1157.yml [hunk 5]
```diff
@@ -126,8 +116,9 @@ discovered:
     may leave the entries blank except for "answer", BUT please write down
     where you looked in "answer".
   answer: |
-    This issue was reported to the TomCat security team by a third party.
-  date: 2009-12-31
+    This vulnerability was discovered by Deniz Cevik, however, after searching through
+    bug reports and CVE statements, I was not able to find an exact date for this discovery.
+  date:
   automated:
   contest:
```
Comment (kwc2894) [inline cves/CVE-2010-1157.yml]: Check me on this but werent automated and contest supposed to be filled either true or false?

File: cves/CVE-2010-1157.yml [hunk 6]
```diff
@@ -138,8 +129,8 @@ subsystem:
     directory names. Look at comments in the code. Look at the bug reports how
     the bug report was tagged.
     Examples: "clipboard", "gpu", "ssl", "speech", "renderer"
-  answer: Taken from directory name
-  name: authenticator
+  answer: The mistake occurred within an authentication component.
```
Comment (kwc2894) [inline cves/CVE-2010-1157.yml]: Was there a specific file involved? (File path?)

**Review by kwc2894 — COMMENTED** [review]
Here is my review (kwc2894)

File: cves/CVE-2010-1157.yml (hunk 1 above)
Comment (rxv3774) [reply cves/CVE-2010-1157.yml]: Updated

File: cves/CVE-2010-1157.yml (hunk 2 above)
Comment (rxv3774) [reply cves/CVE-2010-1157.yml]: Provided an explanation.

File: cves/CVE-2010-1157.yml (hunk 3 above)
Comment (rxv3774) [reply cves/CVE-2010-1157.yml]: Updated

File: cves/CVE-2010-1157.yml (hunk 4 above)
Comment (rxv3774) [reply cves/CVE-2010-1157.yml]: Specified the header.

File: cves/CVE-2010-1157.yml (hunk 5 above)
Comment (rxv3774) [reply cves/CVE-2010-1157.yml]: Updated

File: cves/CVE-2010-1157.yml (hunk 6 above)
Comment (rxv3774) [reply cves/CVE-2010-1157.yml]: Specified file path.

**Pull request merged**